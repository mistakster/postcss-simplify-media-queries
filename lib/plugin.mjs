import { simplifyRules } from "./simplifyRule.mjs";

/**
 * A PostCSS plugin to remove redundant parts of the media description
 */
const plugin = (opts = {}) => {
  return {
    postcssPlugin: 'postcss-simplify-media-queries',
    AtRule(rule) {
      const result = simplifyRules(rule.params);

      if (result.replace(/\s/g, '') !== '(min-width:0)') {
        rule.params = result;
        return;
      }

      rule.replaceWith(rule.nodes);
    }
  }
};

plugin.postcss = true;

export default plugin;
