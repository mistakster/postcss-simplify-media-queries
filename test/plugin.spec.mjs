import { it, expect } from 'vitest';
import postcss from 'postcss';
import plugin from "../lib/plugin.mjs";

function process(css) {
  return postcss([plugin])
    .process(css, { from: undefined })
    .then(result => result.css);
}

it('should work', async () => {
  // language=CSS
  const css = `
      @media (min-width: 200px) and (min-width: 400px) {
          .example {
              padding: 0;
          }
      }
  `;

  await expect(process(css)).resolves.toMatchSnapshot();
});

it('should process (min-width: 0)', async () => {
  // language=CSS
  const css = `
      @media (min-width: 200px) and (min-width: 0) {
          .example {
              padding: 0;
          }
      }
  `;

  await expect(process(css)).resolves.toMatchSnapshot();
});

it('should completely remove useless (min-width: 0)', async () => {
  // language=CSS
  const css = `
      @media (min-width: 0) {
          .example {
              padding: 0;
          }
      }
  `;

  await expect(process(css)).resolves.toMatchSnapshot();
});

it('should parse multiple media queries', async () => {
  // language=CSS
  const css = `
      @media screen and (-ms-high-contrast: none), (-ms-high-contrast: active) {
          .example {
              padding: 0;
          }
      }
  `;

  await expect(process(css)).resolves.toMatchSnapshot();
});
