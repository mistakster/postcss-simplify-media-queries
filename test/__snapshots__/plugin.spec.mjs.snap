// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should completely remove useless (min-width: 0) 1`] = `
"
          .example {
              padding: 0;
          }
  "
`;

exports[`should process (min-width: 0) 1`] = `
"
      @media (min-width:200px) {
          .example {
              padding: 0;
          }
      }
  "
`;

exports[`should work 1`] = `
"
      @media (min-width:400px) {
          .example {
              padding: 0;
          }
      }
  "
`;
